ui/src/kapacitor/containers/TickscriptPage.js

```diff
@@ -23,6 +27,63 @@ class TickscriptPage extends Component {
       },
       validation: '',
       isEditingID: true,
+      logs: [],
+      areLogsEnabled: false,
+    }
+  }
+
+  shouldFetch = null
+
+  fetchChunkedLogs = async (kapacitor, ruleID) => {
+    const {notify} = this.props
+
+    try {
+      const response = await getLogStreamByRuleID(kapacitor, ruleID)
+
+      if (response.status === KAPACITOR_LOGS_NOT_FOUND) {
+        this.setState({
+          areLogsEnabled: false,
+        })
+        this.shouldFetch = false
+        notify(
+          'warning',
+          'Could not use logging, requires Kapacitor version 1.4'
+        )
+        return
+      }
+
+      this.setState({
+        areLogsEnabled: true,
+      })
+      this.shouldFetch = true
```

Comment:
It appears that the state of `areLogsEnabled` and the class property `shouldFetch` mirror each other.  Perhaps use `shouldFetch` as the prop `areLogsEnabled` to the `TickScript` component below?

Reply:
Let me think about that.